tools/topology/topology2/include/pipelines/cavs/chain-dma-playback.conf

```diff
@@ -0,0 +1,128 @@
+#
+# CAVS chained DMA playback pipeline
+#
+# A chain DMA pipeline. All attributes defined herein are namespaced
+# by alsatplg to "Object.Pipeline.passthrough-playback.N.attribute_name"
+#
+# Usage: passthrough-playback pipeline object can be instantiated as:
+#
+# Object.Pipeline.chain-dma-playback."N" {}
+#
+# Where N is the unique pipeline ID within the same alsaconf node.
+#
+
+<include/components/copier.conf>
+<include/components/pipeline.conf>
+<include/common/audio_format.conf>
+
+Class.Pipeline."chain-dma-playback" {
+
+	DefineAttribute."index" {}
+
+	<include/pipelines/pipeline-common.conf>
+
+	attributes {
+		!constructor [
+			"index"
+		]
+
+		!immutable [
+			"direction"
+		]
+
+		#
+		# passthrough-playback objects instantiated within the same alsaconf node must have
+		# unique pipeline_id attribute
+		#
+		unique	"instance"
+	}
+
+	Object.Widget {
+		copier."1" {
+			copier_type	"host"
+			node_type $HDA_HOST_OUTPUT_CLASS
+			type	"aif_in"
+			num_audio_formats 3
+			# 16-bit input format 48KHz 2ch. Output sample format is always 32-bit for playback
+			Object.Base.audio_format.1 {
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+			# 24-bit 48KHz 2ch
+			Object.Base.audio_format.2 {
+				in_bit_depth		32
+				in_valid_bit_depth	24
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$obs * 2]"
+			}
+			# 32-bit 48KHz 2ch
```

Comment:
the frequency and number of channels is N/A, right?

Reply:
@plbossart all copier params are irrelevant for chain DMA but they are here only for the purpose of minimizing the topology work on the driver side.

Reply:
why do we need to add all these if they are ignored? Can we just not add them?

Reply:
if we don't add them, we would have to have to deal with mode of if (!chaind_dma) clauses to deal with in the driver

Reply:
understood, but then why not add one format and add a comment that this is added for compatibility but gets ignored?

Alternatively we could also have a token that says the formats are irrelevant. That's how the Android HAL used to work, there were 'DYNAMIC' formats.

Reply:
sure let me prune these down to the base minimum

Reply:
fixed now @plbossart